### ISSUE TITLE
KeepAliveEnabled property should control "Connection" header, even when the former value is "true".

### ISSUE THREAD

**Issue by flyingpie** [dotnet/wcf#5423]
**Describe the bug**
Currently, when setting the "KeepAliveEnabled"-property to "false", the HttpClient is instructed to send the "Connection"-header with value "close".
When "KeepAliveEnabled" is set to "true" (the default), no "Connection"-header is sent.

Though, according to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection), the default value for the "Connection"-header is "close", so if you don't provide any, the server assumes the connection may be closed after the request.

**To Reproduce**
Set up any binding that uses HTTP (such as basicHttp or basicHttps), don't touch "KeepAliveEnabled", and execute a request. Using a tool like Fiddler, the request can be observed to not have a "Connection"-header.

**Expected behavior**
When setting "KeepAliveEnabled" to "true" (which is the default), a "Connection"-header should be sent, with value "keep-alive". This is the behavior that .NetFx WCF exhibited.

**Screenshots**
Here's a screenshot of a serialized request, with .NetFx on the left, and .NetCore on the right, with the same configuration:
![image](https://github.com/dotnet/wcf/assets/1295673/6598419d-4c45-407d-b361-2c7b3f7f2c3a)

**Additional context**
Here's the code that (as far as I can tell) handles the current behavior:
https://github.com/dotnet/wcf/blob/07f0d0c43119763b505be80d3219a86c8b70e547/src/System.ServiceModel.Http/src/System/ServiceModel/Channels/HttpChannelFactory.cs#L348C1-L351C18

So basically, I'm missing something in there along these lines:

```cs
if (_keepAliveEnabled)
{
    httpClient.DefaultRequestHeaders.Connection.Add("keep-alive");
}
```

Hopefully I've got the point across, looking forward to hear your opinion.

**Comment by mconnew** [comment]
You misread/misunderstood the MDN documentation you linked to, it says the following (***emphasis added***):

> [Directives](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection#directives)
> [`close`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection#close)
> Indicates that either the client or the server would like to close the connection. This is the default on HTTP/1.0 requests.
> 
> [any comma-separated list of HTTP headers [Usually `keep-alive` only]](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection#any_comma-separated_list_of_http_headers_usually)
> Indicates that the client would like to keep the connection open. Keeping a connection open is the default on HTTP/1.1 requests. The list of headers are the name of the header to be removed by the first non-transparent proxy or cache in-between: these headers define the connection between the emitter and the first entity, not the destination node.

The piece you quoted from was for HTTP/1.0. I'm not aware of any SOAP stack which only uses HTTP/1.0, but if it did, HTTP/1.0 doesn't support keep-alive anyway so the header would get ignored either way. HTTP/1.1 is the section you need to pay attention to, where the default is keep-alive. We only need to set the header if we're doing something other than the default.  

We actually have end to end tests verifying this behavior so I'm certain it's working as designed. The headers you listed for NetFx is for HttpWebRequest which has different behavior than HttpClient on .NET [Core] so you can't do a straight comparison. The only way to compare would be to capture the requests with something like fiddler and provide the on-the wire captures showing the connection is being closed.

**Comment by flyingpie** [comment]
@mconnew You're totally right, the combo of HTTP/1.0 instead of HTTP/1.1 and .NetFx sending the "Connection"-header all the time threw me off.

Thank you for your clarification, I'll close this issue :)
